### PR TITLE
feat: Use new AM SDK API

### PR DESCRIPTION
**Pull request by HazAT**
This bumps the JS SDK to the new beta using the new AM API. https://github.com/getsentry/sentry-javascript/pull/2600

### CODE REVIEW

File: build-utils/sentry-instrumentation.js [hunk 1]
```diff
@@ -93,10 +92,8 @@ class SentryInstrumentation {
                 size: `${Math.round(sizeInKb)} KB`,
               },
             });
-            span.startTimestamp = start;
-            span.finish();
-            span.timestamp = start + sizeInKb / 1000;
-            transaction.finish(true);
+            span.finish(start + sizeInKb / 1000);
```
Comment (HazAT) [inline build-utils/sentry-instrumentation.js:95]: @billyvg I made the changes to the file to use the new API.
But this line here in terms of calculating the `endtimestamp` for the Span seems very odd.
Time + Filesize? What is the reason behind this?
I suppose you are abusing the Span chart to show file size?

File: src/sentry/static/sentry/app/bootstrap.tsx [hunk 2]
```diff
@@ -86,10 +86,10 @@ Sentry.init({
...
   integrations: getSentryIntegrations(hasReplays),
   tracesSampleRate,
-  _experiments: {useEnvelope: true},
-  async beforeSend(event) {
-    return normalizeTransactionName(appRoutes, event);
-  },
+});
+
+Sentry.addGlobalEventProcessor(async event => {
+  return normalizeTransactionName(appRoutes, event);
```
Comment (HazAT) [inline src/sentry/static/sentry/app/bootstrap.tsx:92]: @dashed Transaction still goes through event processors and this is the internal workaround for it.
EventProcessors are considered internal.

File: build-utils/sentry-instrumentation.js (hunk 1 above)
Comment (billyvg) [reply build-utils/sentry-instrumentation.js:95]: Correct

File: src/sentry/static/sentry/app/bootstrap.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/bootstrap.tsx:92]: Oh Interesting 👍

**Review by dashed — APPROVED** [review]
Changes for `src/sentry/static/sentry/app/bootstrap.tsx` looks good to me 👍

**Comment by HazAT** [comment]
@dashed Are the percy diffs flakey as always wrt to the performance page?

**Comment by dashed** [comment]
@HazAT ah that might be something we'll have to address. 👍 I approved the Percy snapshots.

**Comment by HazAT** [comment]
Will merge this tomorrow morning

**Review by billyvg — APPROVED** [review]

**Pull request merged**